Source/Plugins/Core/com.equella.core/js/tsrc/modules/Heartbeat.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * The Apereo Foundation licenses this file to you under the Apache License,
+ * Version 2.0, (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { API_BASE_URL } from "../AppConfig";
+import * as OEQ from "@openequella/rest-api-client";
+
+/**
+ * Check oEQ server status by sending a heartbeat request.
+ * If the result is not equal to "OK", throw an error.
+ */
+export function checkHeartbeat() {
+  OEQ.Heartbeat.checkHeartbeat(API_BASE_URL)
+    .then((result: string) => {
+      if (result !== "OK") {
+        throw new Error(
```

Comment:
Since this appears to be a form of type check would it make sense to throw a `TypeError`?
```suggestion
        throw new TypeError(
```

Reply:
Hmmmm, this is not so much validating the type, it's more so validating a business layer response. Ideally it could probably return something like `OK` when all in order, or `UNAVAILABLE` if server is undergoing some kind of maintenance mode etc.

But what is also interesting is that currently the server (if it responds at all) will only ever respond with `OK` - so we could even choose to skip this for now.